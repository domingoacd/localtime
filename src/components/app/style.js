import styled from "styled-components";

export const AppContainer = styled.div`
  position: relative;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 2rem 3rem;
  background-size: cover;
  background-image: url(${(props) => props.bgImage});
`;
