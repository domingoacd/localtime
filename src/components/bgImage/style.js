import styled from "styled-components";

export const Image = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-size: cover;
  background-image: url(${(props) => props.url});
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
`;
