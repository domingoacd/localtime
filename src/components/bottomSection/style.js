import styled from "styled-components";

export const Section = styled.section`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  line-height: 0.9;
`;

export const WelcomeText = styled.p`
  font-size: ${({ theme }) => theme.fontSize.medium};
  text-transform: uppercase;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const LocationText = styled.p`
  font-size: ${({ theme }) => theme.fontSize.medium};
  font-weight: bold;
  text-transform: uppercase;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const LeftSide = styled.div``;

export const RightSide = styled.div``;
