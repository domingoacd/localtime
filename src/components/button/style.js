import styled from "styled-components";

export const Btn = styled.button`
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 0.5rem 0.7rem;
  text-transform: uppercase;
  border-radius: 5rem;
  border: none;
  cursor: pointer;
  color: ${({ theme }) => theme.day.color.secondary};
  background-color: ${({ theme }) => theme.day.color.primary};
`;

export const Arrow = styled.span`
  display: block;
  width: 1rem;
  height: 1rem;
  transform: ${(props) => (props.rotate ? "rotate(180deg)" : "rotate(0deg)")};
  transition: transform 0.5s;
  background-image: url(${(props) => props.image});
  background-size: 100% 100%;
`;
