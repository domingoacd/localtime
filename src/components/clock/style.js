import styled from "styled-components";

export const Time = styled.p`
  font-weight: bold;
  font-size: clamp(2rem, 25vw, 20rem);
  color: ${({ theme }) => theme.day.color.primary};
`;
