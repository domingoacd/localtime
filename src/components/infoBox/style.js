import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  gap: 20vw;
  padding: 3rem;
  margin: 0 -${({ theme }) => theme.spacing.xPadding};
  background-color: rgba(255, 255, 255, 0.7);
`;

export const Col = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5rem;
`;

export const InfoElement = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.4rem;
  line-height: 0.8;
`;

export const InfoName = styled.p`
  font-weight: bold;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.fontSize.small};
  color: ${({ theme }) => theme.day.color.third};
`;

export const Info = styled.h3`
  font-weight: bold;
  font-size: ${({ theme }) => theme.fontSize.big};
  color: ${({ theme }) => theme.day.color.secondary};
`;
