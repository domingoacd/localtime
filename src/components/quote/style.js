import styled from "styled-components";

export const Blockquote = styled.blockquote`
  position: relative;
  display: flex;
  flex-direction: column;
  font-size: ${({ theme }) => theme.fontSize.small};
  visibility: ${(props) => (props.show ? "visible" : "hidden")};
  opacity: ${(props) => (props.show ? "1" : "0")};
  transition: visibility 0.5s, opacity 0.5s;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const QuoteAuthor = styled.span`
  font-weight: bold;
  margin-top: 1rem;
`;
