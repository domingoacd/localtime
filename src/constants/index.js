export const QUOTE_DATA = {
  quote:
    "Donâ€™t postpone joy until you have learned all of your lessons. Joy is your lesson.",
  author: "Alan Cohen",
};
