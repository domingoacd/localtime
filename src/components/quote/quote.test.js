import React from "react";
import renderer from "react-test-renderer";
import Quote from "../quote";
import { ThemeProvider } from "styled-components";
import { theme } from "../../style";
import { QUOTE_DATA } from "../../constants";

test("Renders Quote component", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Quote quoteData={QUOTE_DATA} />
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
