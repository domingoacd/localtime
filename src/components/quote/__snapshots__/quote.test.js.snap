// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Quote component 1`] = `
<blockquote
  className="sc-bdnxRM gBQhwI"
>
  "Donâ€™t postpone joy until you have learned all of your lessons. Joy is your lesson."
  <span
    className="sc-gtsrHT kyKqlw"
  >
    Alan Cohen
  </span>
</blockquote>
`;
